### DISCUSSION TITLE
Empty blocks and Text blocks

### DISCUSSION THREAD

**Discussion by tbmuc** [ImageMagick/ImageMagick#2091] [Help]
Hello,

I would life to create a little complex template out of simple blocks and append them whith the +append command. Its easy to do this with images, but how can I create 
a) empty blocks and 
b) blocks containing only text

of a specific size?

Thomas

**Comment by fmw42** [comment]
To create text, use label: or caption:. They will create a new image with the text.

`convert -size WxH -background white -gravity center caption:"This is some text" text_block.png`

To create empty blocks. Use 

`convert -size WxH xc:white white_block.png`

or

`convert text_block.png -fill white -colorize 100 white_block.png`


Then append them 

`convert white_block.png text_block.png white_block.png -append result.png`

You can do it all in one command as (unix syntax)

```
convert -size WxH -background white -gravity center caption:"This is some text" \
\( +clone -fill white -colorize 100 \) \
\( -clone \) \
-swap 0,1 \
-append \
result.png
```

If using IM 7, then change convert to magick.


Please always identify your IM version and platform, since syntax may vary.

**Reply by tbmuc** [reply]
Thank you very much. How can I center the text into a white block of a specific size?

**Reply by tbmuc** [reply]
The text should be into a separate block

**Comment by fmw42** [comment]
```
convert -size WxH -background white -gravity center caption:"This is some text" \
-extent WWxHH
```

WxH is the limits of the text block.  WWxHH is the size of the white background page into which the text is centered.

For novices, see:

http://www.imagemagick.org/discourse-server/viewtopic.php?f=1&t=9620
http://www.imagemagick.org/script/command-line-processing.php
https://imagemagick.org/script/command-line-options.php
http://www.imagemagick.org/Usage/reference.html
http://www.imagemagick.org/Usage/
https://github.com/ImageMagick/usage-markdown
https://imagemagick.org/script/porting.php#cli
